src/main/scala/io/epiphanous/flinkrunner/serde/ConfluentAvroRegistryKafkaRecordDeserializationSchema.scala

```diff
@@ -76,11 +84,9 @@ class ConfluentAvroRegistryKafkaRecordDeserializationSchema[
     valueDeserializer
       .deserialize(topic, record.value()) match {
       case a: GenericRecord        =>
-        val obj = fromKV(
-          EmbeddedAvroRecordInfo(a.asInstanceOf[A], key, headers)
-        )
+        val obj = toEmbeddedAvroInstance[E, A, ADT](a, avroClass)
```

Comment:
This is where the key change is. We convert the avro record into an instance of our event type, using the `AvroUtils.toEmbeddedAvroInstance()` method. The rest of the changes are formatting or fixed tests.